.github/workflows/build.yml

```diff
@@ -102,15 +103,103 @@ jobs:
             --timeout=1800 \
             --timeout-method=thread \
             -o faulthandler_timeout=1860 \
+            -m "not parallel" \
             -n 12 --dist worksteal \
-            --junit-xml=firedrake.xml \
+            --junit-xml=firedrake1.xml \
             -sv tests
-        timeout-minutes: 120
+        timeout-minutes: 60
+      - name: Test Firedrake 2 ranks
+        run: |
+          . ../firedrake_venv/bin/activate
+          echo OMP_NUM_THREADS is "$OMP_NUM_THREADS"
+          echo OPENBLAS_NUM_THREADS is "$OPENBLAS_NUM_THREADS"
+          mpispawn -nU 12 -nW 2 --propagate-errcodes pytest \
+            --splitting-algorithm least_duration \
+            --splits \$MPISPAWN_NUM_TASKS \
+            --group \$MPISPAWN_TASK_ID1 \
+            --timeout=1800 \
+            --timeout-method=thread \
+            -o faulthandler_timeout=1860 \
+            --junit-xml=firedrake2_\$MPISPAWN_TASK_ID1.xml \
+            -m "parallel[\$MPISPAWN_WORLD_SIZE] and not broken" \
+            -v tests
```

Comment:
TODO: "dogfood" (bleh) `Makefile` and use a matrix to massively cut down on boilerplate